### PR TITLE
[READY] Fixes some clown car bugs and adds emag abilities

### PR DESCRIPTION
**Pull request by ghost**
:cl: Floyd / Qustinnus
add: Clown cars can now fit any mob (besides megafauna)
add: Repair your clown car with bananas
add: Emag the clowncar to unlock a button panel. Activate it to press a random button for a random effect!
balance: lowers health and cost of clown car
fix: removes a return in the clown car code that caused the wrong flags to be assigned
fix: you cant open the clowncar trunk from the inside anymore, you can still escape though.
fix: fixes broken to_chat in clown car
/:cl:

fixes #39943

Once you emag the clown car, you will get a button that activates one out of 6 random abilities on the clown car.

1.  banana peel
2. random chem foam
3. your car turns into singulo (visually only)
4. superlaughter smoke
5. oil trail
6. a fart that makes everyone laugh
https://streamable.com/zqk4w (cooldown is reduced by 66% here to make it easier to showcase)

not all of them are useful, but they are funny.

also
https://youtu.be/R8aGSXCpBR8

### CODE REVIEW

**Comment by fluffe9911** [comment]
Is there anyway to repair your clown car TM

**Comment by Qustinnus** [comment]
no, i guess i could make it so feeding it bananas would fix it.

**Comment by Qustinnus** [comment]
just waiting for sprites now

**Comment by Ziiro** [comment]
Some emag ideas:

* It gives the captives all permanent clown voices

* Activatable Teleport armor effect (read as: bluespace banana peel wax. Think of it like a fighting game counter. Have to activate it, and it is only up for 3 seconds. Triggered or not, it's down for 40 seconds between uses)

* A gilded bike horn radio.

* Add ejecting a random person (except for the clown) to that button you added

**Comment by BeeSting12** [comment]
@Qustinnus The clown car spews out a ton of space lube!/span>/span>

Saw this in the chat today does this PR fix that?

**Comment by Ziiro** [comment]
Some other stuff I've seen that needs to be fixed:
* Hitting it with a non-damaging thing still procs lube
* Passengers inside the car and beat on it

**Comment by Qustinnus** [comment]
@Ziiro the second part is fixed the first i didnt know about

**Comment by Qustinnus** [comment]
@BeeSting12 and yes it does

**Comment by Ziiro** [comment]
* can pick up a pAI'd medibot and the medibot can't be released unless it escapes (crashed multiple times, didn't eject it) (same with manually loaded station pets)
* it's gendering people it sucks up wrong (might have to 
* running it into the supermatter doesn't leave anything behind. Not even ashes

**Comment by Grazyn** [comment]
Saw the car pick up Ian through the Clown pulling the dog inside, but when it crashed seconds later, dog was nowhere to be found

**Comment by Qustinnus** [comment]
Hitting it with a non-damaging thing still procs lube
Passengers inside the car and beat on it
can pick up a pAI'd medibot and the medibot can't be released unless it escapes (crashed multiple times, didn't eject it) (same with manually loaded station pets)
when the car is destroyed in a non-SM way, anyone left inside is deleted

are all fixed. would like a merge for this and then work on the other issues in another PR as to not leave this open for weeks.

**Comment by PKPenguin321** [comment]
Can you make superlaughter actually cause the laughing sound effect the same way regular laughter does

**Comment by Qustinnus** [comment]
it does

**Comment by PKPenguin321** [comment]
are you sure? [maybe im just not seeing it](https://github.com/tgstation/tgstation/blob/bdf3fa3894cd41fd74d1eb8a1fd6fee762ddfca1/code/modules/reagents/chemistry/reagents/drink_reagents.dm#L181) but it looks like it doesnt

**Comment by Qustinnus** [comment]
Oh sorry I mustve confused it with the fart effect. What causes the laughing? is it just the emote for laugh?

**Comment by Qustinnus** [comment]
its gender specific, if I change it I'll do it in a seperate PR because its not necesarily linked to this PR.

**Review by ninjanomnom — CHANGES_REQUESTED** [review]
Seems mostly fine other than the excessive istypes

**Comment by Qustinnus** [comment]
@ninjanomnom done

File: code/modules/vehicles/vehicle_actions.dm [hunk 1]
```diff
@@ -153,3 +153,14 @@
 /datum/action/vehicle/sealed/DumpKidnappedMobs/Trigger()
 	vehicle_entered_target.visible_message("<span class='danger'>[vehicle_entered_target] starts dumping the people inside of it.</span>")
 	vehicle_entered_target.DumpSpecificMobs(VEHICLE_CONTROL_KIDNAPPED)
+
+
+/datum/action/vehicle/sealed/RollTheDice
+	name = "Press a colorful button"
+	desc = "Press one of those colorful buttons on your display panel!."
```
Comment (Carbonhell) [inline code/modules/vehicles/vehicle_actions.dm]: Nice dot there

Comment (Qustinnus) [reply code/modules/vehicles/vehicle_actions.dm]: AAAAAAAAAaaaaaa

Comment (Ziiro) [reply code/modules/vehicles/vehicle_actions.dm]: 🤡

**Pull request merged**